src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts

```diff
@@ -89,14 +91,35 @@ export class ActionTriggerProducerKeyBindingService implements KeyBindingService
     this.actionTriggersWithAction.clear()
 
     this.actionTriggers.forEach(actionTrigger => {
-      const actionForTrigger: Tv2Action | undefined = this.actions.find(action => action.id === actionTrigger.actionId)
-      if (!actionForTrigger) {
+      let action: Tv2Action | undefined = this.actions.find(action => action.id === actionTrigger.actionId)
+      if (!action) {
         return
       }
-      this.actionTriggersWithAction.set(actionTrigger, actionForTrigger)
+      if (action.type === PlaceholderActionType.CONTENT) {
+        action = this.findActionForContentPlaceholderAction(action as Tv2ContentPlaceholderAction, actionTrigger)
+      }
+      if (!action) {
+        return
+      }
+      this.actionTriggersWithAction.set(actionTrigger, action)
     })
   }
 
+  private findActionForContentPlaceholderAction(placeholderAction: Tv2ContentPlaceholderAction, actionTrigger: ActionTrigger<KeyboardTriggerData>): Tv2Action | undefined {
+    const onAirSegment: Segment | undefined = this.rundown?.segments.find(segment => segment.isOnAir)
+    if (!onAirSegment) {
+      return
+    }
+    const actionsForOnAirSegment: Tv2Action[] = this.actions
+      .filter(action => Math.floor(action.rank) === onAirSegment.rank && placeholderAction.metadata.allowedContentTypes.includes(action.metadata.contentType))
```

Comment:
Finding the actions related to the on air segment should be based on the `scope` attribute for the placeholder.

Reply:
Updated. Introduced `getActionFilterCallbackForPlaceholderAction()`